python/cudf/cudf/core/column/string.py

```diff
@@ -652,26 +652,14 @@ def serialize(self):
         frames = []
         sub_headers = []
 
-        sbuf = rmm.device_array(self.nvstrings.byte_count(), dtype="int8")
-        obuf = rmm.device_array(len(self.nvstrings) + 1, dtype="int32")
-        mask_size = utils.calc_chunk_size(
-            len(self.nvstrings), utils.mask_bitsize
-        )
-        nbuf = rmm.device_array(mask_size, dtype="int8")
-        self.nvstrings.to_offsets(
-            libcudf.cudf.get_ctype_ptr(sbuf),
-            libcudf.cudf.get_ctype_ptr(obuf),
-            nbuf=libcudf.cudf.get_ctype_ptr(nbuf),
-            bdevmem=True,
-        )
-        for item in [sbuf, obuf]:
+        for item in reversed(self.children):
```

Comment:
Used `reversed` because it goes offsets and then strings. However deserialization expects the opposite.

Reply:
Can we just make both serialization and deserialization consistent with `self.children` order? We made sure to follow libcudf order.

Reply:
Makes sense. Changed in the latest commit.